In [16]:
from lab8_utils import (
    get_latest_data,
    load_append,
    load_trunc,
    load_incremental,
    show_table,
)
print("import success")

import success


In [19]:
import importlib
import lab8_utils
importlib.reload(lab8_utils)
print(lab8_utils.DB_FILE)

DB_FILE inside module = /Users/yirange/computing/yiran-emily-api/lab_08/lab8.duckdb
/Users/yirange/computing/yiran-emily-api/lab_08/lab8.duckdb


In [22]:
load_append("data/PCPI24M1.csv", init=True)
load_trunc("data/PCPI24M1.csv")
load_incremental("data/PCPI24M1.csv", init=True)

In [23]:
load_append("data/PCPI25M2.csv")
load_trunc("data/PCPI25M2.csv")
load_incremental("data/PCPI25M2.csv")

In [24]:
show_table("cpi_append").tail()
show_table("cpi_trunc").tail()
show_table("cpi_inc").tail()

,date,cpi
932,2024:09,314.9
933,2024:10,315.6
934,2024:11,316.4
935,2024:12,317.6
936,2025:01,319.1


In [25]:
latest_2025 = get_latest_data("data/PCPI25M2.csv").reset_index(drop=True)

append_df = show_table("cpi_append").reset_index(drop=True)
trunc_df = show_table("cpi_trunc").reset_index(drop=True)
inc_df = show_table("cpi_inc").reset_index(drop=True)

print("append matches latest:", append_df.equals(latest_2025))
print("trunc matches latest:", trunc_df.equals(latest_2025))
print("inc matches latest:", inc_df.equals(latest_2025))
print("row counts:", len(append_df), len(trunc_df), len(inc_df), len(latest_2025))

append matches latest: False
trunc matches latest: True
inc matches latest: True
row counts: 976 937 937 937


The results show that the three loading methods behave differently after revisions are introduced.

- `append` does not match the latest source file because it only adds new rows and does not overwrite revised historical values.
- `trunc` matches the latest source file because it reloads the full table each time.
- `incremental` also matches the latest source file because it updates changed rows and inserts new rows.

The row counts confirm this difference:
- `cpi_append`: 976 rows
- `cpi_trunc`: 937 rows
- `cpi_inc`: 937 rows
- latest source data: 937 rows

This shows that `append` kept outdated records, while `trunc` and `incremental` correctly reflected the latest CPI data.